server/src/features/document_symbols.rs

```diff
@@ -384,4 +390,78 @@ impl DocumentSymbolFeature {
         }
     }
 
+///////////////////////////////////////////////
+// XML
+///////////////////////////////////////////////
+
+    fn visit_xml_document(session: &mut SessionInfo, document: roxmltree::Document, results: &mut Vec<DocumentSymbol>, file_info: &Rc<RefCell<FileInfo>>) {
+        let mut children = vec![];
+        for node in document.root_element().children() {
+            if node.is_element() {
+                DocumentSymbolFeature::visit_xml_node(session, &node, &mut children, file_info);
+            }
+        }
+        let range = Range {
+            start: file_info.borrow().offset_to_position(document.root_element().range().start),
+            end: file_info.borrow().offset_to_position(document.root_element().range().end),
+        };
```

Comment:
Style suggestion (DRY): Would be nice to do a function that takes file info, and TextRange and returns Range, it will help as it is duplicated a lot in the code